[TanJieHaoAmos] iP

TanJieHaoAmos opened the pull request:
# Runny ChatBot
> "To be or not to be", William Shakespeare

Runny ChatBot helps you keep your tasks up to date. It's,
* Easy to use
* Text-based
* ~~FAST~~ _SUPER FAST_ to learn

All you need to do is,
1. Download from the [link](https://github.com/TanJieHaoAmos/ip/releases/download/A-Jar/runny.jar.jar)
2. Double-click the file or run the command `java -jar runny.jar.jar` 
3. Add your tasks
4. Let it manage all your tasks. :star_struck:

Most importantly, it is **FREEEE**

Features:
[x] Managing Tasks
[ ] Managing Deadlines (Coming Soon)
[ ] Reminders (Coming Soon)

If you are a Java programmer, you can use it to practice Java too. Here's the main method inside the Runny class:
```java
public static void main(String[] args) {
        new Runny("./data/duke.txt").run();
}
```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

TanJieHaoAmos committed:
Rename main class

TanJieHaoAmos committed:
Update main class

TanJieHaoAmos committed:
Add Main.java
Update AmosTalky.java

TanJieHaoAmos committed:
Update Main.java

TanJieHaoAmos committed:
Complete Level 2
Update Main.java
Update AmosTalky.java

TanJieHaoAmos committed:
Complete Level 3
Add Task.java
Update Main.java
Update AmosTalky.java

TanJieHaoAmos committed:
Complete Level-4
Add Deadline.java
Add Event.java
Add ToDo.java
Update AmosTalky,java
Update Main.java
Update Task.java

TanJieHaoAmos committed:
Update AmosTalky.java

TanJieHaoAmos committed:
Complete A-TextUiTesting
Rename Main.java

TanJieHaoAmos committed:
Complete Level 5
Add RunnyException.java

TanJieHaoAmos committed:
Complete Level-6
Add A-Collections
Update A-TextUiTesting

TanJieHaoAmos committed:
Complete Level-7
Complete A-MoreOOPG
Complete A-Packages

Refactor code to implement Packages and MoreOOPG

TanJieHaoAmos committed:
Merge branch 'branch-Level-7'

* branch-Level-7:
  Complete Level-7 Complete A-MoreOOPG Complete A-Packages

TanJieHaoAmos committed:
Complete Level-8

Add date and time understanding to Runny Chatbot

TanJieHaoAmos committed:
Merge branch 'branch-Level-8'

* branch-Level-8:
  Complete Level-8

TanJieHaoAmos committed:
Merge branch 'add-gradle-support'

TanJieHaoAmos committed:
Add Gradle support

TanJieHaoAmos committed:
Complete A-JUnit

TanJieHaoAmos committed:
Update JUnit testcase

TanJieHaoAmos committed:
Merge branch 'A-JUnit'

* A-JUnit:
  Update JUnit testcase
  Complete A-JUnit

TanJieHaoAmos committed:
Complete A-Jar

TanJieHaoAmos committed:
Merge branch 'A-Jar'

* A-Jar:
  Complete A-Jar

TanJieHaoAmos committed:
Complete Level-9

TanJieHaoAmos committed:
Complete A-CodingStandard

TanJieHaoAmos committed:
Merge branch 'branch-Level-9'

* branch-Level-9:
  Complete Level-9

TanJieHaoAmos committed:
Merge branch 'A-CodingStandard'

* A-CodingStandard:
  Complete A-CodingStandard

TanJieHaoAmos committed:
Complete A-JavaDocs

veehz commented on src/main/java/runny/Runny.java:
```diff
@@ -0,0 +1,71 @@
+ * Some things that it can do includes adding todo, events, deadlines.
+ * It can also delete and find tasks as well as mark and unmark tasks.
+ */
+
```
Perhaps this blank line between the comment and the class should be removed?

```suggestion
```

[Link to relevant line in guide](https://se-education.org/guides/conventions/java/intermediate.html#:~:text=No%20blank%20line%20between%20the%20documentation%20block%20and%20the%20method/class.)

veehz commented on src/main/java/runny/Runny.java:
```diff
@@ -0,0 +1,71 @@
+     *
+     * @param filePath The file path for storing task data.
+     */
+
```
Perhaps this blank line between the comment and the method should be removed?

```suggestion
```

[Link to relevant line in guide](https://se-education.org/guides/conventions/java/intermediate.html#:~:text=No%20blank%20line%20between%20the%20documentation%20block%20and%20the%20method/class.)

veehz commented on src/main/java/runny/RunnyException.java:
```diff
@@ -0,0 +1,15 @@
+/**
+ * Exception class for Runny Chatbot errors.
+ */
+
```
Perhaps this blank line between the comment and the class should be removed? I noticed the same issue in several other places too.

```suggestion
```

[Link to relevant line in guide](https://se-education.org/guides/conventions/java/intermediate.html#:~:text=No%20blank%20line%20between%20the%20documentation%20block%20and%20the%20method/class.)

veehz commented on src/main/java/runny/task/Event.java:
```diff
@@ -0,0 +1,81 @@
+     */
+    @Override
+    public String toString() {
+
```
It doesn't violate any coding standards but perhaps removing this blank line will make it nicer?

```suggestion
```

veehz commented on src/main/java/runny/ui/Ui.java:
```diff
@@ -0,0 +1,56 @@
+    /**
+     * Displays a welcome message to the user.
+     */
+    public void welcomeMessage() {
```
Should this be a verb instead?
```suggestion
    public void printWelcomeMessage() {
```

[Link to Relevant Line in Guide](https://se-education.org/guides/conventions/java/intermediate.html#:~:text=Names%20representing%20methods%20must%20be%20verbs%20and%20written%20in%20camelCase.)

veehz commented on src/main/java/runny/ui/Ui.java:
```diff
@@ -0,0 +1,56 @@
+    /**
+     * Displays an exit message to the user.
+     */
+    public void byeMessage() {
```
Should this be a verb instead?
```suggestion
    public void printByeMessage() {
```

[Link to Relevant Line in Guide](https://se-education.org/guides/conventions/java/intermediate.html#:~:text=Names%20representing%20methods%20must%20be%20verbs%20and%20written%20in%20camelCase.)

veehz reviewed (commented):
Overall, I find your code easy to understand as it has descriptive comments. I noticed a few JavaDoc-blank-lines-related coding standard violations.

TanJieHaoAmos committed:
Refactor code to add GUI to Runny Chatbot

TanJieHaoAmos committed:
Merge branch 'branch-Level-10'

TanJieHaoAmos committed:
Add assertions to the code at various points.

Assertions are used to define assumptions about the program state so that the runtime can verify them. By doing this, we can make sure that everything is running as intended.

By adding specific outputs for assertions, we can easily identify the areas with bugs.

TanJieHaoAmos committed:
Fix code to improve coding quality

TanJieHaoAmos committed:
Merge pull request #1 from TanJieHaoAmos/branch-A-Assertions

Add assertions to the code at various points

TanJieHaoAmos committed:
Merge branch 'master' of https://github.com/TanJieHaoAmos/ip

* 'master' of https://github.com/TanJieHaoAmos/ip:
  Add assertions to the code at various points.

TanJieHaoAmos committed:
Merge pull request #2 from TanJieHaoAmos/branch-A-CodeQuality

Fix code to improve coding quality

TanJieHaoAmos committed:
Merge branch 'master' of https://github.com/TanJieHaoAmos/ip

* 'master' of https://github.com/TanJieHaoAmos/ip:
  Fix code to improve coding quality

TanJieHaoAmos committed:
Refactor code for BCD-Extension

TanJieHaoAmos committed:
Merge branch 'branch-BCD-Extension'

* branch-BCD-Extension:
  Refactor code for BCD-Extension

TanJieHaoAmos committed:
Fix wrong string output in DeadlineCommand.java

TanJieHaoAmos committed:
Improve GUI

Allow window to be resize, with objects following the resized window

TanJieHaoAmos committed:
Add A-Personality

TanJieHaoAmos committed:
Add files via upload

Upload Ui.png into docs folder

TanJieHaoAmos committed:
Fix output messages and GUI Window name

TanJieHaoAmos committed:
Merge branch 'master' of https://github.com/TanJieHaoAmos/ip

* 'master' of https://github.com/TanJieHaoAmos/ip:
  Add files via upload

TanJieHaoAmos committed:
Update README.md

TanJieHaoAmos committed:
Fix bug in DateTime for Event task

TanJieHaoAmos committed:
Fix CheckStyle issue
